geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/Stat.scala

```diff
@@ -0,0 +1,65 @@
+/***********************************************************************
+  * Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+  * All rights reserved. This program and the accompanying materials
+  * are made available under the terms of the Apache License, Version 2.0 which
+  * accompanies this distribution and is available at
+  * http://www.opensource.org/licenses/apache2.0.php.
+  *************************************************************************/
+
+package org.locationtech.geomesa.utils.stats
+
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.util.parsing.combinator.RegexParsers
+
+trait Stat {
+  def observe(sf: SimpleFeature)
+  def add(other: Stat): Stat
+
+  def toJson(): String
+}
+
+object Stat {
+
+  class StatParser extends RegexParsers {
```

Comment:
This'll get exposed as WPS, and as someone calls that, I could imagine wanting a little DSL to specify desired stats.